app/models/donation.rb

```diff
@@ -0,0 +1,92 @@
+class Donation
+  include ActiveModel::Model
+
+  attr_accessor :email, :employer, :occupation, :stripe_token, :recurring,
+    :utm_source, :utm_medium, :utm_campaign, :source_url, :amount_in_cents
+
+  validates :email, presence: true,
+    email_format: { message: 'is not a valid email' }
+  validates :employer, presence: true
+  validates :occupation, presence: true
+  validates :stripe_token, presence: true
+  validates :amount_in_cents, presence: true,
+    numericality: { greater_than: 0, only_integer: true }
+
+  def process!
+    if valid?
+      process_payment
+      record_donation
+      create_donate_action
+    else
+      false
+    end
+
+  rescue Stripe::CardError => e
+    errors.add(:stripe_token, e.json_body[:error][:message])
+    false
+  end
+
+  private
+
+  def process_payment
+    if recurring
+      stripe_customer = StripeCustomer.new(create_stripe_customer)
+      person.update(stripe_id: stripe_customer.id)
+      person.create_subscription(remote_id: stripe_customer.subscription_id)
+    else
+      create_stripe_charge
+    end
+  end
+
+  def record_donation
+    person = { email: email, employer: employer, occupation: occupation }
+    NbDonationCreateJob.perform_later(amount_as_integer, person)
+  end
+
+  def create_donate_action
+    person.create_action(utm_source: utm_source,
+                         utm_medium: utm_medium,
+                         utm_campaign: utm_campaign,
+                         source_url: source_url,
+                         template_id: 'donate',
```

Comment:
One note here: `tempalte_id` should be able to be overridden from the params sent to the backend - for tracking data purposes.